internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/pending.go

```diff
@@ -135,13 +135,34 @@ func (h *Handler) AcceptReceivedShare(w http.ResponseWriter, r *http.Request) {
 // RejectReceivedShare handles DELETE Requests on /apps/files_sharing/api/v1/shares/{shareid}
 func (h *Handler) RejectReceivedShare(w http.ResponseWriter, r *http.Request) {
 	shareID := chi.URLParam(r, "shareid")
-	data := h.updateReceivedShare(w, r, shareID, true, "")
+	fieldMask := &fieldmaskpb.FieldMask{Paths: []string{"rejectshare"}}
+	data := h.updateReceivedShare(w, r, shareID, "", fieldMask)
 	if data != nil {
 		response.WriteOCSSuccess(w, r, []*conversions.ShareData{data})
 	}
 }
 
-func (h *Handler) updateReceivedShare(w http.ResponseWriter, r *http.Request, shareID string, rejectShare bool, mountPoint string) *conversions.ShareData {
+func (h *Handler) UpdateReceivedShare(w http.ResponseWriter, r *http.Request) {
+	shareID := chi.URLParam(r, "shareid")
+	fieldMask := &fieldmaskpb.FieldMask{Paths: []string{"hidden"}}
+	data := h.updateReceivedShare(w, r, shareID, "", fieldMask)
+	if data != nil {
+		response.WriteOCSSuccess(w, r, []*conversions.ShareData{data})
+	}
+}
+
+func (h *Handler) updateReceivedShare(w http.ResponseWriter, r *http.Request, shareID, mountPoint string, fieldmask *fieldmaskpb.FieldMask) *conversions.ShareData {
+	var rejectShare, hideFlag bool
+	for i := range fieldmask.Paths {
+		switch fieldmask.Paths[i] {
+		case "rejectshare":
+			rejectShare = true
```

Comment:
As just discussed in person setting a fixed value here kinda violates the idea behind the fieldmask. A possible refactoring would be to pass a `ReceivedShare` object instead of the separate attributes and use that in the `UpdateReceivedShareRequest` instead of handcrafting one.